xdsl/parser.py

```diff
@@ -688,7 +687,7 @@ def _parse_optional_block_label(
         A block label consists of block-id ( `(` block-arg `,` ... `)` )?
         """
         block_id = self.try_parse_block_id()
-        arg_list = list()
+        arg_list = list[tuple[Span, Attribute]]()
```

Comment:
What is the difference to `arg_list : List[...] = []`? Does this thing just do type inference instead of explicit annotatios?

Reply:
it does for local variables